core/src/test/java/org/testcontainers/utility/AuthenticatedImagePullTest.java

```diff
@@ -153,6 +151,18 @@ public void testThatAuthLocatorIsUsedForDockerComposePull() throws IOException {
         }
     }
 
+    private Path getLocalTempFile(String s) throws IOException {
+        Path projectRoot = Paths.get(".");
+        Path tempDirectory = Files.createTempDirectory(projectRoot, this.getClass().getSimpleName() + "-test-");
+        Path relativeTempDirectory = projectRoot.relativize(tempDirectory);
+        Path tempFile = Files.createTempFile(relativeTempDirectory, "test", s);
+
+        tempDirectory.toFile().deleteOnExit();
```

Comment:
IIRC Java does not support deleting non-empty directories. Let's change the order (file first, directory later) to at least give it a chance to empty :)

Reply:
I'm not sure it cares (no errors evident, and nothing gets left over), but sure :)

Reply:
AFAIK @bsideup is right about this, but on the other hand, it should throw `IOException` if it fails to delete. Weird...

Reply:
TL;DR, counterintuitively the original form of the code was OK.